src/compositions/usePagination.ts

```diff
@@ -123,29 +127,27 @@ export function usePagination<
     page.value--
   }
 
-  function setFetchSubscriptionParameters(): void {
-    fetchSubscriptionParameters.value = getFetchSubscriptionParameters()
-  }
-
-  function getFetchSubscriptionParameters(): Parameters<TFetch>[] | null {
-    const parameters = fetchParametersGetter()
-
-    if (parameters === null) {
-      return null
+  function getPagesToFetch(page: number): number[] {
+    if (mode === 'page') {
+      return [page]
     }
 
-    const [filter, ...rest] = parameters
-    const filters = getFetchParametersForPages(page.value, filter)
-
-    return filters.map(filter => [filter, ...rest]) as unknown as Parameters<TFetch>[]
+    return repeat(page, index => index + 1)
   }
 
-  function getFetchParametersForPages(page: number, filter?: TFetchFilter): TFetchFilter[] {
-    if (mode === 'page') {
-      return [getFetchFilterForPage(page, filter)]
-    }
+  function getFetchParametersForPage(page: number): Ref<Parameters<TFetch> | null> {
+    return toRef(() => {
+      const parameters = fetchParametersGetter()
 
-    return repeat(page, index => getFetchFilterForPage(index + 1, filter))
+      if (parameters === null) {
+        return null
+      }
+
+      const [filter, ...rest] = parameters
+      const pageFilter = getFetchFilterForPage(page, filter)
+
+      return [pageFilter, ...rest]
+    }) as Ref<Parameters<TFetch> | null>
```

Comment:
I'm surprised this needs to be cast, what's up there?

Reply:
Common mistake with our subscriptions and compositions. They're expecting `Ref<T>` but a wrapped getter produces `Readonly<Ref<T>>` so typescript complains about the ref being readonly. Something that would be good to fix at some point but it rarely comes up.

Reply:
got it, thanks